samples/apps/copilot-chat-app/webapi/CopilotChat/Controllers/DocumentImportController.cs

```diff
@@ -76,108 +76,182 @@ private enum SupportedFileType
     /// Service API for importing a document.
     /// </summary>
     [Authorize]
-    [Route("importDocument")]
+    [Route("importDocuments")]
     [HttpPost]
     [ProducesResponseType(StatusCodes.Status200OK)]
     [ProducesResponseType(StatusCodes.Status400BadRequest)]
-    public async Task<IActionResult> ImportDocumentAsync(
+    public async Task<IActionResult> ImportDocumentsAsync(
         [FromServices] IKernel kernel,
         [FromServices] IHubContext<MessageRelayHub> messageRelayHubContext,
         [FromForm] DocumentImportForm documentImportForm)
     {
-        var formFile = documentImportForm.FormFile;
-        if (formFile == null)
+        try
         {
-            return this.BadRequest("No file was uploaded.");
+            await this.ValidateDocumentImportFormAsync(documentImportForm);
         }
+        catch (ArgumentException ex)
+        {
+            return this.BadRequest(ex.Message);
+        }
+
+        this._logger.LogInformation("Importing {0} document(s)...", documentImportForm.FormFiles.Count);
 
-        if (formFile.Length == 0)
+        // TODO: Perform the import in parallel.
```

Comment:
This is potential bug in VolatileMemoryStore that is prevent multiple tasks from writing to a new collection. Will come work on the optimization when we figure it out.

Reply:
I observed that inserting memories to VolatileMemoryStore is not multi-thread safe. If you have two tasks that are inserting memories to a collection that doesn't exist yet, VolatileMemoryStore will sometimes fail at creating the collection on one of the tasks because of name conflicts (https://github.com/microsoft/semantic-kernel/blob/main/dotnet/src/SemanticKernel/Memory/VolatileMemoryStore.cs#L25). 

We can remove the condition in `VolatileMemoryStore.CreateCollectionAsync` since `TryAdd` will only return `false` when a key already exists, which is probably fine in our use case. However, I am uncertain if this will just create more issues. I talked to the SK core team, and they said it's not recommended to use SK in a multithreaded app. Need further conversation.

Reply:
Might be worth putting a comment in summarizing this

Reply:
Will create a backlog item instead.